rs/sns/governance/src/governance.rs

```diff
@@ -696,6 +700,12 @@ pub struct Governance {
     /// The number of proposals after the last time "garbage collection" was run.
     pub latest_gc_num_proposals: usize,
 
+    /// Global lock for all periodic tasks that relate to upgrades - this guarantees
```

Comment:
How is that guaranteed?

Reply:
I updated the comment to say this:

> Global lock for all periodic tasks that relate to upgrades - this is used to guarantee that they don't interleave with one another outside of rare circumstances (e.g. timeouts).

Do you think the exact mechanism requires further explanation?